LabDrivers/utils.py

```diff
@@ -743,11 +744,35 @@ def close(self):
         #self.reset()
         pass
 
-    def control_ren(self, level):
+    def control_ren(self, level, addr = None):
+        if level is None:
+            return
+        ASSERT_REN = [1,3]
+        DEASSERT_REN = [0,2]
+        GTL = [2,6]
+        LLO = [4,5]
+
+        ADDR_DEV = [3,4]
         logging.debug("Currently, control_ren only asserts local mode")
-        if level == 0 or True: # TODO: CHANGE `OR TRUE`
-            self.write('++loc')
+        #### DO ASSERTS #####
+        if level in ASSERT_REN:
+            self.assertren = True
+            logging.warning("Unabled to assert REN on prologix controller")
+        elif level in DEASSERT_REN:
+            self.assertren = False
+            logging.warning("Unabled to deassert REN on prologix controller")
+        if level in GTL:
+            if level in ADDR_DEV and addr is not None:
+                self.write('++loc %d'%addr)
```

Comment:
I would add a reference to the manual where the ++loc command is described

Reply:
I found a page for both ++loc and ++llo command and added it to the documentation